### PR TITLE
docs: Update modtap docs and README.md

### PR DESCRIPTION
**Pull request by okke-formsma**
Updates the mod-tap docs to be correct and explain about the relation between mod-tap and hold-tap.

Updates the README.md with more links to the docs and more confident language.

### CODE REVIEW

**Review by adamcstephens — APPROVED** [review]

**Comment by innovaker** [comment]
Please run `prettier` and repush 👍   I personally recommend the VS Code extension.

**Comment by innovaker** [comment]
Could we perhaps start with a short reference to `hold-tap`, in addition to the juicier note at the bottom?  For instance ...
> _Mod-Tap_ is a special instance of _[Hold-Tap](http://zmkfirmware.dev/docs/behavior/hold-tap)_.

File: docs/docs/behavior/mod-tap.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ sidebar_label: Mod-Tap
 
 ## Summary
 
-The Mod-Tap sends a different keypress, if it's tapped or held. When you tap the key shortly, the first keycode is sent. If you hold the key for longer than 200ms, the second keycode is sent.
+The Mod-Tap sends a different keypress, depending on if it's held or tapped. If you hold the key for longer than 200ms, the first keycode ("mod") is sent. When you tap the key shortly, the second keycode ("tap") is sent.
```
Comment (innovaker) [inline docs/docs/behavior/mod-tap.md:8]: Suggestion to improve readability ...
```suggestion
The Mod-Tap behavior sends a different keypress, depending on whether it's held or tapped.
- If you hold the key for longer than 200ms, the first keycode ("mod") is sent.
- If you tap the key (release before 200ms), the second keycode ("tap") is sent.
```

**Comment by innovaker** [comment]
@okke-formsma, it looks like 2871dca9bfce2588e23da0905d7142ef5ff1c1a1 has affected this PR and GitHub isn't handling it gracefully.  Please could you rebase onto `HEAD` and force push.  Thanks!

**Comment by okke-formsma** [comment]
Rebased.

**Review by petejohanson — APPROVED** [review]
Thanks!

**Pull request merged**